exporter/awsemfexporter/datapoint.go

```diff
@@ -273,25 +375,72 @@ func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int,
 		bucketBegin = -math.Pow(base, float64(index+1))
 		metricVal := (bucketBegin + bucketEnd) / 2
 		count := negativeBucketCounts.At(i)
-		if count > 0 {
-			arrayValues = append(arrayValues, metricVal)
-			arrayCounts = append(arrayCounts, float64(count))
+		if count > 0 && firstSplit.length < firstSplit.maxBuckets {
+			firstSplit.values = append(firstSplit.values, metricVal)
+			firstSplit.counts = append(firstSplit.counts, float64(count))
+			firstSplit.length++
+			currentLength++
+			if firstSplit.maxBuckets < totalBucketLen {
+				firstSplit.count += int(count)
+				if currentLength == firstSplit.length {
+					firstSplit.bucketMin = bucketEnd
+					if bucketBegin < bucketEnd {
+						firstSplit.bucketMin = bucketBegin
+					} else {
+						firstSplit.bucketMin = bucketEnd
+					}
+				}
+			}
+		} else if count > 0 {
+			if currentLength == firstSplit.maxBuckets && currentLength != 0 {
+				if bucketBegin < bucketEnd {
+					secondSplit.bucketMax = bucketEnd
+				} else {
+					secondSplit.bucketMax = bucketBegin
+				}
+			}
+			secondSplit.values = append(secondSplit.values, metricVal)
+			secondSplit.counts = append(secondSplit.counts, float64(count))
+			currentLength++
 		}
 	}
 
-	return []dataPoint{{
+	//fmt.Println("firstSplit", firstSplit)
+	//fmt.Println("secondSplit", secondSplit)
+
+	var datapoints []dataPoint
+	// Add second data point (last 100 elements or fewer)
+	datapoints = append(datapoints, dataPoint{
 		name: dps.metricName,
 		value: &cWMetricHistogram{
-			Values: arrayValues,
-			Counts: arrayCounts,
-			Count:  metric.Count(),
+			Values: firstSplit.values,
+			Counts: firstSplit.counts,
+			Count:  uint64(firstSplit.count),
 			Sum:    metric.Sum(),
-			Max:    metric.Max(),
-			Min:    metric.Min(),
+			Max:    firstSplit.bucketMax,
+			Min:    firstSplit.bucketMin,
 		},
 		labels:      createLabels(metric.Attributes()),
 		timestampMs: unixNanoToMilliseconds(metric.Timestamp()),
-	}}, true
+	})
+
+	// Add first data point with the remaining elements
+	if totalBucketLen > 100 {
```

Comment:
Seems like this should be 
```suggestion
	if totalBucketLen > splitThreshold {
```

Reply:
Updated